xio-core/src/main/java/com/xjeffrose/xio/tracing/HttpTracingState.java

```diff
@@ -1,38 +1,19 @@
 package com.xjeffrose.xio.tracing;
 
 import brave.Span;
-import brave.propagation.TraceContext;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.util.AttributeKey;
+import java.util.Optional;
 
 public class HttpTracingState {
 
   private static final AttributeKey<Span> span_key = AttributeKey.newInstance("xio_tracing_span");
-  private static final AttributeKey<TraceContext> context_key =
-      AttributeKey.newInstance("xio_tracing_context");
 
   public static void setSpan(ChannelHandlerContext ctx, Span span) {
     ctx.channel().attr(span_key).set(span);
   }
 
-  public static Span getSpan(ChannelHandlerContext ctx) {
-    return ctx.channel().attr(span_key).get();
-  }
-
-  public static boolean hasSpan(ChannelHandlerContext ctx) {
-    return ctx.channel().attr(span_key).get() != null;
-  }
-
-  public static Span popSpan(ChannelHandlerContext ctx) {
-    return ctx.channel().attr(span_key).getAndSet(null);
-  }
-
-  public static void setContext(ChannelHandlerContext ctx, TraceContext context) {
-    ctx.channel().attr(context_key).set(context);
-  }
-
-  // get the context set by outbound request XioRequestEncoder (outbound)
-  public static TraceContext popContext(ChannelHandlerContext ctx) {
-    return ctx.channel().attr(context_key).getAndSet(null);
+  public static Optional<Span> popSpan(ChannelHandlerContext ctx) {
+    return Optional.ofNullable(ctx.channel().attr(span_key).getAndSet(null));
   }
 }
```

Comment:
now that we have the tracing information in the request/response objects we should consider removing this concept of attaching it to the `ctx`.

Reply:
yeah - i'm just not sure how we would handle `HttpClientResponseTracingHandler.exceptionCaught` since we do not have a reference to Response